script/cibuild

```diff
@@ -0,0 +1,5 @@
+#!/usr/bin/env bash
+
+set -e
+
+. script/test
```

Comment:
``` caba@caba-ThinkPad-T450s /tmp % cat test-main.sh 
#!/bin/bash

echo "Main process pid: $$"

. child-code.sh
source ./child-code.sh

./child-code.sh
eval ./child-code.sh
caba@caba-ThinkPad-T450s /tmp % cat child-code.sh 
#!/bin/bash

echo "Included code pid: $$"
caba@caba-ThinkPad-T450s /tmp % ./test-main.sh   
Main process pid: 8409
Included code pid: 8409
Included code pid: 8409
Included code pid: 8410
Included code pid: 8411
```

In this specific case, I'd opt for creating a subshell for the tests themselves (so ./script/test), rather than just including the tests code in the cibuild (that is what actually happens when you run . script/test or include script/test). Reason is the tests are an autonomous unit, that may call "exit", or they can be run in background, and they should not share also the variables in the scope of cibuild... I normally limit the usage of "include" or ". script" to variables definition, functions that support the main code...

Reply:
good point!